regression-tests/mixed-type-safety-1.cpp2

```diff
@@ -26,12 +27,12 @@ print: ( msg: std::string, b: bool ) =
 
 main: () -> int = 
 {
-    print( "1.1 is int? ", 1.1 is int );
-    print( "1   is int? ", 1   is int );
+    ::print( "1.1 is int? ", 1.1 is int );
```

Comment:
Explicit specification of the global namespace fixes an ambiguity issue caused by [std::print](https://en.cppreference.com/w/cpp/io/print).

Reply:
I was trying to figure out why the cpp1 compiler considers `std::print` in this call... it looks like that compiler on some platforms makes them globally accessible.

Reply:
I also found it strange, esp. that Clang-18 (C++23) was failing only for `mixed-type-safety-1.cpp2`, while MSVC (latest) was failing only for `pure2-type-safety-1.cpp`...

Reply:
Ah... right. The argument is from `std` namespace. I forgot about this rule. 

@gregmarr Thank you!

Reply:
Yes, I though it was ADL, as I rwrote in the PR description, but I found it strange there was come inconsistency in how differently the recent compiles (MSVC, GCC 14 and Clang-18) behave for `mixed-type-safety-1.cpp2` on current main.

Reply:
Yes, the inconsistency is strange.  Do all of those compilers have `std::print` available with the compiler options that we are using?  Otherwise they must just have different disambiguation rules.

Reply:
Thanks for figuring this out! This was on my list to look into and I'm glad to see it's been figured out, much appreciated.

Reply:
Should there be a comment there about why the `::` is needed, so someone doesn't try to remove it in the future?

Should the function just be renamed?

Can we somehow make the user-facing end result of that `is` be `bool` even when it's the compile-time `std::true_type` or `std::false_type` so that users don't end up with these same ADL issues?  Would that defeat the purpose of using those types in the first place?  If so, do the benefits of returning that type outweigh the cost?